src/dispatchers/browserTypeDispatcher.ts

```diff
@@ -47,4 +50,29 @@ export class BrowserTypeDispatcher extends Dispatcher<BrowserType, channels.Brow
       defaultContext: browser._defaultContext ? new BrowserContextDispatcher(browserDispatcher._scope, browser._defaultContext) : undefined,
     };
   }
+
+  async connect(params: channels.BrowserTypeConnectParams): Promise<channels.BrowserTypeConnectResult> {
+    const waitForNextTask = params.slowMo
+      ? (cb: () => any) => setTimeout(cb, params.slowMo)
+      : makeWaitForNextTask();
+    const paramsHeaders = Object.assign({'User-Agent': getUserAgent()}, params.headers || {});
+    const ws = new WebSocket(params.wsEndpoint, [], {
+      perMessageDeflate: false,
+      maxPayload: 256 * 1024 * 1024, // 256Mb,
+      handshakeTimeout: params.timeout || 30000,
+      headers: paramsHeaders,
+    });
+    const pipe = new JsonPipeDispatcher(this._scope);
+    ws.on('open', () => pipe.wasOpened());
+    ws.on('close', () => pipe.wasClosed());
+    ws.on('error', error => pipe.wasClosed(error));
+    pipe.on('close', () => ws.close());
+    pipe.on('message', message => ws.send(JSON.stringify(message)));
```

Comment:
- Perhaps we should make the pipe send strings instead of objects to avoid extra parse/stringify?
- I'd try/catch here and send an error if something went wrong. For example, we had to check ws.readyState in some places, otherwise it throws.

Reply:
- In JS there is no extra parse/stringify atm and for other languages, I don't think we can do anymore much more clever while staying a proxy
- So it'll throw in the driver event handler, that is probably ok when the socket is closed?